### DISCUSSION TITLE
Changing the time range doesn't update the panel

### DISCUSSION THREAD

**Discussion by psheehymn** [gapitio/gapit-htmlgraphics-panel#146] [Q&A] [answered]
We are using the HTML Graphics visualization to display a table of real user monitoring (RUM) metrics due to the greater control it allows on the formatting over the base Grafana Table visualization. However, when the user changes the time range on the dashboard, the HTML Graphics panel does not update; the data being displayed is for the previous time range. Ditto for a Custom variables that allow the user to select the percentile (P75, P95) on the load time measurement. Both require a browser-level refresh to get the panel to update.

Is there a setting to get the panel to refresh when the dashboard time range or a variable changes?

**Comment by ZuperZee** [comment]
I would imagine that the code updating the values in the table is in the onInit code without using the [panelupdate event](https://gapit-htmlgraphics-panel.gapit.io/docs/references/#panelupdate-event) instead of the onRender code.

It's tough to debug without an example or something minimal which shows the problem.

The normal code when creating an HTMLGraphics code shows how to update variables when the dashboard time range changes or variable changes.

**Comment by psheehymn** [comment, marked as answer]
I overlooked the panelupdate event. That did the trick.
